lib/tasks/upgrade_wiki_articles.rake

```diff
@@ -0,0 +1,101 @@
+  values = YAML.safe_load(File.read(Rails.root.join("config/arrays/wiki", "values.yml")))
+  defaults = YAML.safe_load(File.read(Rails.root.join("config/arrays/wiki", "defaults.yml")))
+
+  user = User.find_by_username("mitsiee")
+  user = User.find_by_username("admin") unless @user.present?
+
+  (actions + values).each do |current|
+    article = Wiki::Article.where(title: current["en-US"]).last
+
+    next if article.blank?
+    next if article.content.include?("### Properties")
+
+    content_before_arguments = article.content.split("### Arguments").first.strip
+    content_without_return = content_before_arguments.split("### Return").first.strip
+    description_only = content_without_return.gsub("### Description", "").strip
+
+    parameters_short = (current["args"] || []).map { |item| "==_#{ item["name"] }_==" }.join(", ")
+    parameters_snippet = "(#{ (current["args"] || []).map { |item| defaults[item["default"]] || item["default"] }.join(", ") })"
+    parameters_block = (current["args"] || []).map { |item|
+      type_array = item["type"].is_a?(String) ? [item["type"] || "Void"] : item["type"]
+      type_string_array = []
+
+      type_array.map { |type|
+        type_article = Wiki::Article.where(title: type).last if type && type.is_a?(String)
+        type_slug = type_article.present? ? type_article.slug : nil
+
+        if type.is_a?(String)
+          type_string_array.push("**#{ type_slug.present? ? "[#{ type }](/wiki/articles/#{ type_slug })" : (type || "Void") }**")
+        else
+          type_string_array.push("**#{ type[0] || type.keys[0] }** (**#{ type[1] || type.values[0] }**)")
+        end
+      }
+"
+> ==**#{ item["name"] }**==
+> _Type: #{ type_string_array.join(" | ") }, Default: **#{ item["default"] }**_
+> #{ item["description"] }
+".strip
+    }.join("\n\n")
+
+    return_array = current["return"].is_a?(String) ? [current["return"]] : current["return"]
+    return_string = current["return"].blank? ? "_<span style=\"color: #c678dd\">Void</span>_" : return_array.map { |item|
+      return_value_is_string = item.is_a?(String)
+      return_value_article = Wiki::Article.where(title: item).last if item.present? && return_string
+      return_value_slug = return_value_article.present? ? return_value_article.slug : nil
+
+      if item.is_a?(String)
+        "_<span style=\"color: #c678dd\">#{ return_value_slug.present? ? "[#{ item }](/wiki/articles/#{ return_value_slug })" : (item || "Void") }</span>_"
+      else
+        "_<span style=\"color: #c678dd\">#{ item[0] || item.keys[0] }</span> (<span style=\"color: #c678dd\">#{ item[1] || item.values[0] }</span>)_"
+      end
+    }.join(" | ")
+
+    content = "
+## Description
+
+#{ description_only == current["description"] ? "<big>#{ description_only }</big>" : description_only }
+
+## Snippet
+
+```
+#{ current["en-US"] }#{ current["args"].present? ? parameters_snippet : "" };
+```
+
+## Properties
```

Comment:
```suggestion
## Arguments
```

Reply:
or

```suggestion
## Parameters
```

(matching what workshop.codes/editor calls them)

Reply:
This is the general header for the properties of the action, which includes both the return value and parameters

Reply:
Ah I see. Didn't scroll further down 👍

Reply:
I went with parameters over arguments. They are used interchangeably, but they don't actually mean the same thing. To put it in Javascript:

```
function thing(one, two) { ... } <- Parameters

thing(one, two) <- Arguments
```

The available options in the function are parameters, but you give arguments as values for those parameters.